test/MinimalDelegation.execute.t.sol

```diff
@@ -445,76 +406,86 @@ contract MinimalDelegationExecuteTest is TokenHandler, DelegationHandler, Execut
     /// forge-config: default.isolate = true
     /// forge-config: ci.isolate = true
     function test_execute_native_single_batchedCall_opData_eoaSigner_gas() public {
-        Call[] memory calls = CallBuilder.init();
+        Call[] memory calls = CallUtils.init();
         calls = calls.push(buildTransferCall(address(0), address(receiver), 1e18));
 
-        uint256 nonceKey = 0;
-        (uint256 nonce,) = _buildNextValidNonce(nonceKey);
-        SignedCalls memory signedCalls = SignedCalls({calls: calls, nonce: nonce});
+        bytes memory executionData = abi.encode(calls, _signAndPack(hash(calls, DEFAULT_NONCE), signerTestKey, DEFAULT_NONCE, ROOT_KEY_HASH));
```

Comment:
i think this makes it harder to understand how data is encoded for the contracts imo. like i honestly forget where the nonce goes in the opdata structuring bc there are multiple things being packed right? the signature and then the nonce?

Maybe it would help to just separate out a few steps?

wrappedSignature = _signAndPack(key, signedCalls) // note the nonce is already passed in
opData = pack(nonce, wrappedSignature)
executionData = encode(calls, opData)

Reply:
That makes sense, my only point is that it takes so much additional code to write a passing test right now because of how specific the encoding is (maybe that's a meta point for the contracts lol), but these helper functions were meant to be used in tests where we're not explicitly testing something related to the encoding

Reply:
OK I simplified it!